src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/AsylumCaseCallbackApiDelegator.java

```diff
@@ -49,7 +49,10 @@ public AsylumCase delegate(
         HttpEntity<Callback<AsylumCase>> requestEntity = new HttpEntity<>(callback, setHeaders(serviceAuthorizationToken,accessToken));
 
         try {
-            log.info("Invoking API endpoint: {}", endpoint);
+            long caseId = callback.getCaseDetails().getId();
```

Comment:
Based on unit-tests outcome, I'd say we need to wrap this in a null check.

```suggestion
            long caseId = Optional.of(callback.getCaseDetails()).map(details -> details.getId()).orElse(0L);
```